api/v1alpha7/types.go

```diff
@@ -108,10 +108,11 @@ type PortOpts struct {
 	// The virtual network interface card (vNIC) type that is bound to the neutron port.
 	VNICType string `json:"vnicType,omitempty"`
 
-	// A dictionary that enables the application running on the specified
-	// host to pass and receive virtual network interface (VIF) port-specific
-	// information to the plug-in.
-	Profile map[string]string `json:"profile,omitempty"`
+	// Profile is a set of key-value pairs that are used for binding details.
+	// We intentionally don't expose this as a map[string]string because we only want to enable
+	// the users to set the values of the keys that are known to work in OpenStack Networking API.
+	// See https://docs.openstack.org/api-ref/network/v2/index.html?expanded=create-port-detail#create-port
+	Profile BindingProfile `json:"bindingProfile,omitempty"`
```

Comment:
The public-facing name of this field is the json name, which I wasn't anticipating we would change. Was there a specific reason to change it from `profile` to `bindingProfile`?

If we do decide to change it to `bindingProfile`, please could you also update the Go field name to match?

Reply:
Let's stick to `Profile`. I'll update everything.

Reply:
Also if you don't mind, I'll let the `BindingProfile` for the new struct.